src/link/object.c

```diff
@@ -450,12 +476,29 @@ void obj_ReadFile(char const *fileName)
 		}
 	}
 
+	uint32_t nbAsserts;
+
+	tryReadlong(nbAsserts, file, "%s: Cannot read number of assertions: %s",
+		    fileName);
+	verbosePrint("Reading %u assertions...\n", nbAsserts);
+	for (uint32_t i = 0; i < nbAsserts; i++) {
+		struct Assertion *assertion = malloc(sizeof(*assertion));
+
+		if (!assertion)
```

Comment:
If you're out of memory for such a small allocation, issue a fatal error immediately and `exit()`. You won't be able to do much else — it's very likely that `err()` itself will fail.

Reply:
`err` *is* a fatal error—the `1` argument is the exit status.